homeassistant/components/demo/calendar.py

```diff
@@ -67,6 +67,14 @@ async def async_get_events(
         end_date: datetime.datetime,
     ) -> list[CalendarEvent]:
         """Return calendar events within a datetime range."""
+        if start_date.tzinfo is None:
+            start_date = start_date.replace(
```

Comment:
This is a bug in `list_events` and not something that should be added to the demo.
https://github.com/home-assistant/core/blob/4581c3664879a848da111a2caa7dbeebebd1da71/homeassistant/components/calendar/__init__.py#L581

This logic needs to be moved into `list_events` similar to the other caller 
https://github.com/home-assistant/core/blob/4581c3664879a848da111a2caa7dbeebebd1da71/homeassistant/components/calendar/__init__.py#L581 and it should call `dt_util.as_local`

Reply:
Sent https://github.com/home-assistant/core/pull/95804 to fix the list events bug